src/test/java/LinkedIteratorTest.java

```diff
@@ -11,103 +12,107 @@ public class LinkedIteratorTest {
     private <T> Node<T> makeLinkedStructure(T t1, T... ts) {
         Node<T> head = new Node<>(t1);
         Node<T> current = head;
-        for (int i = 0; i < ts.length; ++i) {
-            Node<T> node = new Node<>(ts[i]);
+        for (T t : ts) {
+            Node<T> node = new Node<>(t);
             current.setNext(node);
             current = node;
         }
         return head;
     }
 
-    @Test
-    void hasNext_sizeZeroLinkedStructure_returnsFalse() {
-        Node<Integer> head = null;
-        Iterator<Integer> it = new LinkedIterator<>(head);
-        assertFalse(it.hasNext());
-    }
+    @Nested
+    class WhenEmpty {
```

Comment:
Similarly with these tests, asserting dumping the iterator into a list then comparing equality on the list is simpler and more complete. Iterators are highly stateful, such a strategy nets you only cardinality cases to test.